source/agora/consensus/data/Transaction.d

```diff
@@ -247,3 +248,44 @@ public bool getSumOutput (const Transaction tx, ref Amount acc)
             return false;
     return true;
 }
+
+/*******************************************************************************
+
+    Caculates the fee of transaction data to store
+
+    Params:
+        data_size = The size of the data
+        factor = The factor to calculate for the fee of transaction data
+
+    Return:
+        A fee payable to data storage
+
+*******************************************************************************/
+
+public Amount clculateDataFee(ulong data_size, uint factor) @safe nothrow
+{
+    const double decimal = 100.0;
+    ulong fee = cast(ulong)(
+        (
+            round(
+                (
+                    exp((cast(double)data_size / cast(double)factor)) -
+                    1.0
+                ) *
+                decimal
+            ) /
+            decimal
+        ) *
+        10_000_000L
+    );
+    return Amount(fee);
+}
+
+unittest
+{
+    // When the size of a data is 0 bytes the fee is 0 BOA
```

Comment:
These are not sufficient tests. Try to test on a range of values. For example using power of two could work.

For example:

```D
clculateDataFee(0, 200);
clculateDataFee(2, 200);
clculateDataFee(4, 200);
clculateDataFee(8, 200);
clculateDataFee(16, 200);
...
```

and so on.. And also with a different factor too.